Game logic Implementation

gdumani opened the pull request:
This milestone's objective is to create the functional TIC TAC TOE game logic.
The game was already working in the previous milestone. We added the following features

- Classes were removed from main.rb and added to a library of classes in lib where user.rb an board.rb were created

maelfosso commented on bin/main.rb:
```diff
@@ -1,62 +1,21 @@
 end
 
-board = Board.new
 3.times { |i| p board.b[i] }
```
That chunk of code appears many time.
Write a method into the `Board` class that will return a **String** which will be print in `main.rb`
```rb
class Board
  ...
  def display
    " ... "
  end
  ...
end
```

Call it like this in `bin/main.rb`
```rb
p board.display
```

maelfosso commented on lib/board.rb:
```diff
@@ -0,0 +1,34 @@
+    @symbols = %w[X O]
+  end
+
+  def turn(symb, loc)
+    if !loc.to_s.match?(/[1-9]/) || loc.to_s.length != 1 || @b[(loc - 1) / 3][(loc - 1) % 3].match?(/[XO]/)
+      @status = false
+    else
+      @status = true
+      @b[(loc - 1) / 3][(loc - 1) % 3] = symb
+      if winner(@b)
+        @counter = 0
+        @symb = symb
+      end
+      @counter -= 1
+    end
+  end
```
That method do many things. Please, separate it
1- Check if the user input is valid
2- Update the Board
3- Checking the winner

Create at least those 3 methods into the `Board` class and update your while loop in `main.rb`.
This will make your code clearer and easily understandable.

gdumani commented:
1. Added a current method to Board class to return the index of the current user directly from the class.
2. Added a display method on Board class that will format the output to with a puts compatible string. We added some colors to the box numbers to make it the board easier to read.
3. The logic doesn't require changes for the winner status. Winner will be false when symb returned by Board at the end of the game is empty, meaning a draw.  Board doesn't store user information other than the symbol. It returns the winning symbol and we do a lookup in user array  to fetch its corresponding name when we print the winner. The elements in array user contain objects generated by the User class with the required information.

serdg0 commented on lib/board.rb:
```diff
@@ -0,0 +1,44 @@
+      end
+      @counter -= 1
+    end
+  end
```
This methods is violating the Single Responsability Principle.
Refer to the previous review, so it should be splitted into 3 methods.

gdumani commented:
Modified turn method and created invalid_input and udate_board methods
